spark/src/main/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpAlter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.flint.operation;
+
+import lombok.SneakyThrows;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.spark.client.EMRServerlessClient;
+import org.opensearch.sql.spark.dispatcher.model.FlintIndexOptions;
+import org.opensearch.sql.spark.execution.statestore.StateStore;
+import org.opensearch.sql.spark.flint.FlintIndexMetadata;
+import org.opensearch.sql.spark.flint.FlintIndexMetadataService;
+import org.opensearch.sql.spark.flint.FlintIndexState;
+import org.opensearch.sql.spark.flint.FlintIndexStateModel;
+
+/**
+ * Index Operation for Altering the flint index. Only handles alter operation when
+ * auto_refresh=false.
+ */
+public class FlintIndexOpAlter extends FlintIndexOp {
+  private static final Logger LOG = LogManager.getLogger(FlintIndexOpAlter.class);
+  private final EMRServerlessClient emrServerlessClient;
+  private final FlintIndexMetadataService flintIndexMetadataService;
+  private final FlintIndexOptions flintIndexOptions;
+
+  public FlintIndexOpAlter(
+      FlintIndexOptions flintIndexOptions,
+      StateStore stateStore,
+      String datasourceName,
+      EMRServerlessClient emrServerlessClient,
+      FlintIndexMetadataService flintIndexMetadataService) {
+    super(stateStore, datasourceName);
+    this.emrServerlessClient = emrServerlessClient;
+    this.flintIndexMetadataService = flintIndexMetadataService;
+    this.flintIndexOptions = flintIndexOptions;
+  }
+
+  @Override
+  protected boolean validate(FlintIndexState state) {
+    return state == FlintIndexState.ACTIVE || state == FlintIndexState.REFRESHING;
+  }
+
+  @Override
+  FlintIndexState transitioningState() {
+    return FlintIndexState.UPDATING;
+  }
+
+  @SneakyThrows
+  @Override
+  void runOp(FlintIndexMetadata flintIndexMetadata, FlintIndexStateModel flintIndexStateModel) {
+    LOG.debug(
+        "Running alter index operation for index: {}", flintIndexMetadata.getOpensearchIndexName());
+    this.flintIndexMetadataService.updateIndexToManualRefresh(
```

Comment:
if auto_refresh is changed to false, but cancelStreamingJob failed, does ALTER INDEX auto_refresh=false retry able?

Reply:
I am making the query retryable by allowing the auto_refresh=false even if existing state is already in auto_refresh=false. So any transient errors in cancelling job can be retried.